modules/REST/V1/Options.php

```diff
@@ -0,0 +1,218 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\REST\V1;
+
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+use WP_Syntex\Polylang\Options\Options as Options_Registry;
+use WP_Syntex\Polylang\REST\Abstract_Controller;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Options REST controller.
+ *
+ * @since 3.7
+ */
+class Options extends Abstract_Controller {
+	/**
+	 * @var Options_Registry
+	 */
+	private $options;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.7
+	 *
+	 * @param Options_Registry $options Options registry.
+	 */
+	public function __construct( Options_Registry $options ) {
+		parent::__construct();
+		$this->rest_base = 'options';
+		$this->options   = $options;
+	}
+
+	/**
+	 * Registers the routes for options.
+	 *
+	 * @since 3.7
+	 *
+	 * @return void
+	 */
+	public function register_routes(): void {
+		register_rest_route(
+			$this->namespace,
+			"/{$this->rest_base}",
+			array(
+				'args'        => array(
+					'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_items' ),
+					'permission_callback' => array( $this, 'update_items_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+				'schema'      => array( $this, 'get_public_item_schema' ),
+				'allow_batch' => array( 'v1' => true ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves all options.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of array
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function get_items( $request ) {
+		return $this->prepare_item_for_response( $this->options->get_all(), $request );
+	}
+
+	/**
+	 * Updates option(s).
+	 * This allows to unpdate one or several options.
+	 * Note: to update an array value, use `option=val1,val2`.
```

Comment:
A JSON body can handle array shape easily, right?

Reply:
Indeed. This is when passed in the URL. It is misleading so I removed it in [c74b5ba](https://github.com/polylang/polylang/pull/1569/commits/c74b5badb0b1df7aec6a3cc74431ae4b3a4dd6df).

Reply:
We shouldn't be aware of that, this is WP business IMHO.
We should only do `$request->get_param( 'foo' )`.

Reply:
???